aioxmpp/presence/service.py

```diff
@@ -438,3 +591,66 @@ def resend_presence(self):
 
         if self.client.established:
             return self.client.enqueue(self.make_stanza())
+
+    def subscribe_peer_directed(self,
+                                peer: aioxmpp.JID,
+                                muted: bool = False):
+        """
+        Create a directed presence relationship with a peer.
+
+        :param peer: The address of the peer. This can be a full or bare JID.
+        :type peer: :class:`aioxmpp.JID`
+        :param muted: Flag to create the relationship in muted state.
+        :type muted: :class:`bool`
+        :rtype: :class:`DirectedPresenceHandle`
+        :return: The new directed presence handle.
+
+        `peer` is the address of the peer which is going to receive directed
+        presence. For each bare JID, there can only exist either a single bare
+        JID directed presence relationship, or zero or more full JID directed
+        presence relationships. It is not possible to have a bare JID directed
+        presence relationship and a full JID directed presence relationship for
+        the same bare JID.
+
+        If `muted` is :data:`False` (the default), the current presence is
+        unicast to `peer` when the relationship is created and the relationship
+        is created with :attr:`~.DirectedPresenceHandle.muted` set to
+        :data:`False`.
+
+        If `muted` is :data:`True`, no presence is sent when the relationship is
+        created, and it is created with :attr:`~.DirectedPresenceHandle.muted`
+        set to :data:`True`.
+
+        If the user of this method needs to set a
+        :attr:`~.DirectedPresenceHandle.presence_filter` on the relationship,
+        creating it with `muted` set to true is the only way to achieve this
+        before the initial directed presence to the peer is sent.
+
+        The newly created handle is returned.
+        """
+
+    def rebind_directed_presence(self,
+                                 relationship: DirectedPresenceHandle,
+                                 new_peer: aioxmpp.JID):
+        """
```

Comment:
When is the rebinding necessary? Do we direct presence to full JIDs? (Note: This is probably just my ignorance.)

Reply:
Yes, we do for MUC. In MUC, you send directed presence to $bare_muc_jid/$nickname. When changing the nickname (either by choice or because the server forces you), you have to redirect the directed presence. Ugly, but that’s how it is.